src/server/decorators/auth.guard.ts

```diff
@@ -0,0 +1,33 @@
+import {
+  ExecutionContext,
+  Injectable,
+} from '@nestjs/common';
+import { Observable } from 'rxjs';
+import { AuthGuard as PassportAuthGuard } from '@nestjs/passport';
+import { ConfigService } from '../services';
+
+/**
+ * Extends Passport's default AuthGuard with a new implementation
```

Comment:
I wonder if this also should have a dummy version that just returns `true` in dev? What do you think?

Reply:
Opted not to do this since it would just move the `if this.devMode` to a custom provider without providing any real visible benefit(as above)